models/default_local_provider.go

```diff
@@ -474,10 +474,7 @@ func (l *DefaultLocalProvider) RemotePatternFile(req *http.Request, resourceURL,
 		owner := parsedPath[1]
 		repo := parsedPath[2]
 		branch := "main"
-
-		if len(parsedPath) == 4 {
-			branch = parsedPath[3]
-		}
+		branch = parsedPath[len(parsedPath)-1]
```

Comment:
Since the user will want to use URL such as: `https://github.com/layer5io/meshery/blob/master/.goreleaser.yml`. 
- URL.Path returns `/owner/repo/blob/master`
- `Split("/")` will return an array of length 5 with the first element is an empty string due to the slash at the beginning

The previous implementation will return `blob` instead of `master` causing an error from importing remotely. Instead, we can just return the last element in the array for correctness.

Let me know your thoughts @utkarsh-pro

Reply:
@Kn99HN actually the endpoint expects the github URL to look like `https://github.com/<owner>/<repo>/[branch]` where the branch is optional (defaults to "main").

In any case we would need to standardise how the endpoint is supposed to look like. The current [design spec](https://docs.google.com/document/d/1B2N78EdRiZF-yVo1-HY3syppwBBDumgMuYg6seD-AJ4/edit#heading=h.blih70a9hxp) standardises the above mentioned format as it captures all relevant information that would be needed to walk github. 

This representation is also similar to how github structures there "raw" data endpoints, which look like `https://raw.githubusercontent.com/<owner>/<repo>/<branch>/...`

Reply:
Ah, that makes sense. It seems that the requirement for the 2 are different. But I'll change it back to your implementation and update it on my end. Thank you!